dts/bindings/reserved-memory/nordic,owned-memory.yaml

```diff
@@ -8,17 +8,63 @@ description: |
   will be recorded in the UICR of the compiled domain. Memory ownership and
   access is then configured for the domain at boot time, based on the UICR.
 
+  Example:
+
+    reserved-memory {
+        memory@2fc00000 {
+            compatible = "nordic,owned-memory";
+            reg = <0x2fc00000 0x1000>;
+            status = "okay";
+            nordic,access = <NRF_OWNER_ID_APPLICATION NRF_PERM(READ, SECURE)>,
+                            <NRF_OWNER_ID_RADIOCORE NRF_PERM(WRITE, SECURE)>;
+        };
+    };
+
+  A single local domain can request a memory region to be reserved on behalf of
+  multiple access owners. A single memory region shall be reserved by at most
+  one domain, by setting status "okay" on the associated node. For example, if
+  the region defined above is enabled by Application on behalf of Radiocore,
+  then the Radiocore's devicetree must set status "disabled" on that node.
+
+  Each of the different owners may have a different set of permissions granted,
+  as also shown above.
+
+  Note: one domain can also reserve memory for another domain and not itself.
+  Whichever domain has status "okay" set on the node does not need to be listed
+  as one of the access owners.
+
 compatible: "nordic,owned-memory"
 
-include: base.yaml
+include: [base.yaml, "zephyr,memory-common.yaml"]
 
 properties:
   reg:
     required: true
 
+  nordic,access:
+    type: array
+    description: |
+      Array of (owner-id, permission-flags) pairs, where:
+
+      - Owner ID represents the domain that will have access to this memory.
+        Valid values can be found in dts/common/nordic/<soc>.dtsi,
+        where they are defined as NRF_OWNER_ID_*
+
+      - Permissions are encoded as a 32-bit bitfield, using the flags found in
+        include/zephyr/dt-bindings/reserved-memory/nordic-owned-memory.h,
+        where they are defined as NRF_PERM_*
+
+      Each pair typically uses the NRF_PERM shorthand macro:
+        <NRF_OWNER_ID_APPLICATION NRF_PERM(READ, EXEC)>
+
+      This expands to:
+        <NRF_OWNER_ID_APPLICATION (NRF_PERM_READ | NRF_PERM_EXEC)>
+
   owner-id:
     type: int
+    deprecated: true
```

Comment:
is it worth deprecating at this stage?

Reply:
Yes, maybe not for Zephyr but for downstream, deprecation is practical in the short-term.